src/helpers.rs

```diff
@@ -905,24 +905,26 @@ impl<'a, 'mir: 'a, 'tcx: 'a + 'mir> CurrentSpan<'a, 'mir, 'tcx> {
     /// This function is backed by a cache, and can be assumed to be very fast.
     pub fn get(&mut self) -> Span {
         let idx = self.current_frame_idx();
-        Self::frame_span(self.machine, idx)
+        self.stack().get(idx).map(Frame::current_span).unwrap_or(rustc_span::DUMMY_SP)
     }
 
-    /// Similar to `CurrentSpan::get`, but retrieves the parent frame of the first non-local frame.
+    /// Similar to `CurrentSpan::get`, but if the last frame is local, we return its parent.
     /// This is useful when we are processing something which occurs on function-entry and we want
     /// to point at the call to the function, not the function definition generally.
+    /// We need to be sure to only back up one frame if the frame we would select is the last
+    /// frame, otherwise we've already backed up to or past the caller in our search for a local
+    /// frame.
```

Comment:
If we keep this comment it should go inside the function -- it explains the implementation, not the specification.